### ISSUE TITLE
Make allocateCmdOptions optional

### ISSUE THREAD

**Issue by sebelik** [chromedp/chromedp#562]
I'm trying to run chromedp with Chromium (79.0.3945.0) in AWS Lambda but when `Run` is called for the first time and chromedp tries to spawn a new browser process, it fails with
```
fork/exec chromium: operation not permitted
```

This is caused by the call to `allocateCmdOptions` in `ExecAllocator.Allocate`, where it tries to set `pdeathsig` on the child Chrome process, which is (probably) not allowed in Lambda environment.

**Would it be possible to make the `allocateCmdOptions` call optional?** For FaaS environments like Lambda it serves no purpose anyway since if the main (chromedp) process dies, the whole environment (including the Chrome browser process) is destroyed and restarted.

**Comment by mvdan** [comment]
Is there a way to detect whether or not the feature is available? Adding more API for this seems wrong, we should just do the right thing on each environment.

**Comment by sebelik** [comment]
I've tried to find a way to detect it but it turns out AWS is using custom virtualization technology,  [Firecracker](https://github.com/firecracker-microvm/firecracker), with its own set of security policies and process isolation tools. 

Without extensive understanding of the technology and the Linux OS itself, it's hard for me to tell how exactly to detect what features are available to a process under Firecracker but this seems to do the job for me:

```go
import (
	"unsafe"

	"golang.org/x/sys/unix"
)

func isJailed() bool {

	var sig int
	
	err := unix.Prctl(unix.PR_GET_PDEATHSIG, uintptr(unsafe.Pointer(&sig)), 0, 0, 0)
	if err != nil {
		return err.Error() == "operation not permitted"
	}

	return false

}
```

**Big disclaimer though**: I know very little about the inner workings of Linux virtualization and process isolation and this may not be the right way to do this at all.

**Comment by mvdan** [comment]
Detecting if the error is "operation not permitted" and ignoring it would be fine. Ideally there would be a better way to check for that than a string comparison, but it doesn't seem like that error has an exported type we can compare against.

If you send a PR I'll be happy to merge it. No test is required, as that would be very hard to do. My only suggestion is that you use `strings.Contains`, so that we do a substring match.

**Comment by cgilling** [comment]
I just ran into this and hacked together a version locally, it should be noted that this wasn't the final fix. I also had to make sure that there were some extra command line flags sent into chrome as well. I just hacked in adding the flags recommended in [this article](https://medium.com/@marco.luethy/running-headless-chrome-on-aws-lambda-fa82ad33a9eb) and was able to get it to work after that. Without doing that I wouldn't get the exec error, but my lambda would spin forever until it timed out

**Comment by yildizfatih** [comment]
> I just ran into this and hacked together a version locally, it should be noted that this wasn't the final fix. I also had to make sure that there were some extra command line flags sent into chrome as well. I just hacked in adding the flags recommended in [this article](https://medium.com/@marco.luethy/running-headless-chrome-on-aws-lambda-fa82ad33a9eb) and was able to get it to work after that. Without doing that I wouldn't get the exec error, but my lambda would spin forever until it timed out

I am having the same problem. Do you mind sharing working command-line options and is there any diff for the change with the working local version? Thanks.

**Comment by cgilling** [comment]
> I am having the same problem. Do you mind sharing working command-line options and is there any diff for the change with the working local version? Thanks.

@yildizfatih 

I pushed the code up to my fork [here](https://github.com/chromedp/chromedp/compare/master...cgilling:lamda_hack?expand=1). There is probably a less hacky way to set these settings, I was just doing this for a personal project and could just do a `replace` in my `go.mod` so didn't really care.

It should also be noted that I found after a while that disk space would be filled up from files in /tmp as well as after a while that connections would just fail. I'd say in each lambda invocation I would make like ~10 calls and have to change the RAM setting every 8-10 invocations to force a clean reset.
